### PR TITLE
Add Fortnet recipe

### PR DESCRIPTION
**Pull request by vanderhe**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by vanderhe** [comment]
@conda-forge-admin, please ping team

**Comment by vanderhe** [comment]
@conda-forge-admin, please ping team

**Comment by vanderhe** [comment]
@conda-forge-admin, please ping team

File: recipes/fortnet/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  license_family: LGPL
+  license_file:
+    - "LICENSE"
+    - "COPYING"
+    - "COPYING.LESSER"
+    - "external/fsockets/LICENSE"
+    - "external/xmlf90/LICENSE"
+  summary: "Fortnet, a software package for training Behler-Parrinello neural networks"
+  doc_url: https://{{ name }}.readthedocs.io
+  dev_url: https://github.com/{{ origin }}/{{ name }}
```
Comment (carterbox) [inline recipes/fortnet/meta.yaml:80]: Please detemplate these urls so they can be easily accessed.

File: recipes/fortnet/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  {% set mpi_prefix = "mpi_" + mpi %}
+  {% else %}
+  {% set mpi_prefix = "nompi" %}
+  {% endif %}
+  string: "{{ mpi_prefix }}_h{{ PKG_HASH }}_{{ build }}"
+
+  {% if mpi != "nompi" %}
+  run_exports:
+    - {{ name }} * {{ mpi_prefix }}_*
+  {% endif %}
```
Comment (carterbox) [inline recipes/fortnet/meta.yaml:32]:
```suggestion
  run_exports:
    - {{ name|lower }} * {{ mpi_prefix }}_*
    - {{ pin_subpackage( name|lower, max_pin='x.x.x') }}
```
Anyone building against this library needs the version pinned also. The max_pin of the API should be set according to ABI guarantees. MAJOR version less than 1 usually means no ABI guarantees, so pin to the PATCH level.

File: recipes/fortnet/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('fortran') }}
+    - python
+    - h5py
+    - numpy
+    - cmake
+    - ninja
+    - fypp
+    - wget  # [build_platform == target_platform]
```
Comment (carterbox) [inline recipes/fortnet/meta.yaml:44]: Why do you need wget? All the source code should come from the hashed release from GitHub or from a conda-forge package.

File: recipes/fortnet/build.sh [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+cmake_options=(
+   ${CMAKE_ARGS}
+   "-DLAPACK_LIBRARY='lapack;blas'"
+   "-DWITH_SOCKETS=ON"
+   "-DWITH_MPI=${MPI}"
+   "-DBUILD_SHARED_LIBS=ON"
+   "-GNinja"
+   ${cmake_mpi_options[@]}
+   ..
+)
```
Comment (carterbox) [inline recipes/fortnet/build.sh:19]:
```suggestion
cmake_options=(
   ${CMAKE_ARGS}
   "-DCMAKE_BUILD_TYPE=Release"
   "-DLAPACK_LIBRARY='lapack;blas'"
   "-DWITH_SOCKETS=ON"
   "-DWITH_MPI=${MPI}"
   "-DBUILD_SHARED_LIBS=ON"
   "-GNinja"
   ${cmake_mpi_options[@]}
   ..
)
```
We want Release, [not ReleasewithDebInfo.](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=598957&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=2071)

File: recipes/fortnet/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    - hdf5 >=1.10.4  # [build_platform != target_platform]
+    - hdf5 * nompi_*  # [build_platform != target_platform]
+  host:
+    - {{ mpi }}  # [mpi != "nompi"]
+    - mpifx * mpi_{{ mpi }}_*  # [mpi != "nompi"]
+    - libblas
+    - liblapack
```
Comment (carterbox) [inline recipes/fortnet/meta.yaml:51]: This package has multiple [warnings about overdepending](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=598957&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=2239). Please remove liblapack if it is not linked (static linking is not allowed).

Also, it's strange that there is a non-link warning about mpich and openmpi too.

**Review by carterbox — CHANGES_REQUESTED** [review]

File: recipes/fortnet/meta.yaml (hunk 1 above)
Comment (vanderhe) [reply recipes/fortnet/meta.yaml]: Thanks for reviewing. I've removed the templates in the urls.

File: recipes/fortnet/meta.yaml (hunk 2 above)
Comment (vanderhe) [reply recipes/fortnet/meta.yaml]: You're right, I've applied your suggestions.

File: recipes/fortnet/meta.yaml (hunk 3 above)
Comment (vanderhe) [reply recipes/fortnet/meta.yaml]: I forgot to remove the wget dependency when copying over the meta.yaml from another project. It's removed now.

File: recipes/fortnet/build.sh (hunk 4 above)
Comment (vanderhe) [reply recipes/fortnet/build.sh]: I see, it's now compiled in Release mode.

File: recipes/fortnet/meta.yaml (hunk 5 above)
Comment (vanderhe) [reply recipes/fortnet/meta.yaml:49]: I've also seen the warnings, but do not understand them, as linking without e.g. liblapack dependency fails (tried again, just to be sure). Same confusion about mpich/openmpi.

**Comment by vanderhe** [comment]
@carterbox I have incorporated all the suggestions of the code review and resolved the overdependencies. Could you take another look at it?

File: recipes/fortnet/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+  run:
+    - {{ pin_compatible('mpifx', max_pin='x.x') }} mpi_{{ mpi }}_*  # [mpi != "nompi"]
+    # hdf5 >=1.10.4 has run_exports
+    - hdf5
+
+test:
+  commands:
+    - export OMPI_MCA_plm=isolated OMPI_MCA_btl_vader_single_copy_mechanism=none OMPI_MCA_rmaps_base_oversubscribe=yes  # [mpi == "openmpi"]
+    - test -f $PREFIX/bin/fnet
+    - $PREFIX/bin/fnet | grep "Error in opening file"  # [mpi == "nompi" and build_platform == target_platform]
```
Comment (carterbox) [inline recipes/fortnet/meta.yaml:57]:
```suggestion
    - fnet | grep "Error in opening file"  # [mpi == "nompi" and build_platform == target_platform]
```
Shouldn't need to use full path.

File: recipes/fortnet/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('fortran') }}
+    - python
+    - h5py
+    - numpy
+    - cmake
+    - ninja
+    - fypp
```
Comment (carterbox) [inline recipes/fortnet/meta.yaml:42]:
```suggestion
    - cmake
    - ninja
```
We're not running any of these tests, so the python libraries are not needed?

**Review by carterbox — CHANGES_REQUESTED** [review]

File: recipes/fortnet/meta.yaml (hunk 7 above)
Comment (vanderhe) [reply recipes/fortnet/meta.yaml]: As far as I can see python, h5py and numpy are needed for my [minimal tests](https://github.com/vanderhe/staged-recipes/blob/00208afb855d2ba739f123d438ea7a4d37c91815/recipes/fortnet/build.sh#L45-L55) to ensure build sanity. However, I removed the fypp dependency.

Comment (carterbox) [reply recipes/fortnet/meta.yaml]: I didn't see those tests! Thanks for pointing them out.

**Pull request merged**